src/jit/lower.cpp

```diff
@@ -2988,10 +2988,12 @@ bool Lowering::AreSourcesPossiblyModified(GenTree* use, GenTree* src1, GenTree*
 //
 void Lowering::LowerAddrMode(GenTreePtr* pTree, GenTree* before, Compiler::fgWalkData* data, bool isIndir)
 {
-    GenTree* addr = *pTree;
-    GenTreePtr  base, index;
-    unsigned    scale, offset;
-    bool        rev;
+    GenTree*    addr   = *pTree;
+    GenTreePtr  base   = nullptr;
+    GenTreePtr  index  = nullptr;
+    unsigned    scale  = 0;
```

Comment:
Since `genCreateAddrMode` couldn't have meaningfully read `index` it seems like we could set the initial value to 1 here and remove the `if (scale == 0)` code at line 3003. But I'm ok leaving it this way if you prefer.

Have you seen any cases where initializing this changes the jit's behavior?